modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/SleepTimer.kt

```diff
@@ -48,16 +49,23 @@ class SleepTimer @Inject constructor(@ApplicationContext private val context: Co
         val alarmManager = getAlarmManager()
         alarmManager.cancel(sleepIntent)
 
-        return try {
-            alarmManager.setExactAndAllowWhileIdle(AlarmManager.RTC_WAKEUP, timeMs, sleepIntent)
-            sleepTimeMs = timeMs
-            true
-        } catch (e: Exception) {
-            LogBuffer.e(LogBuffer.TAG_CRASH, e, "Unable to start sleep timer.")
-            if (Build.VERSION.SDK_INT >= 31 && !alarmManager.canScheduleExactAlarms()) {
-                Toast.makeText(context, LR.string.player_sleep_timer_start_failed, Toast.LENGTH_LONG).show()
-            }
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && !alarmManager.canScheduleExactAlarms()) {
+            Toast.makeText(context, LR.string.player_sleep_timer_start_failed, Toast.LENGTH_LONG).show()
+            context.startActivity(
+                Intent(ACTION_REQUEST_SCHEDULE_EXACT_ALARM)
+                    // Because we're not launching this from an activity context, we must add the FLAG_ACTIVITY_NEW_TASK flag
+                    .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
```

Comment:
Instead of adding this flag we could pass an `Activity` instance (or a `startActivity: (Intent) -> Unit` function from an activity) to this method and then start this intent without the `FLAG_ACTIVITY_NEW_TASK` flag, but I don't think adding the flag will cause any harm in this circumstance and it keeps the code a bit cleaner so I went with this approach. Let me know if you think we're better off avoiding the flag.